runtime/string_functions.cpp

```diff
@@ -1957,15 +1957,29 @@ string f$str_repeat(const string &s, int64_t multiplier) {
   return result;
 }
 
-static string str_replace_char(char c, const string &replace, const string &subject, int64_t &replace_count) {
+static string str_replace_char(char c, const string &replace, const string &subject, int64_t &replace_count, bool with_case) {
   int count = 0;
-  const char *piece = subject.c_str(), *piece_end = subject.c_str() + subject.size();
+  const char *piece = subject.c_str();
+  const char *piece_end = subject.c_str() + subject.size();
+
   string result;
   if (!replace.empty()) {
     result.reserve_at_least(subject.size());
   }
+
   while (true) {
-    const char *pos = (const char *)memchr(piece, c, piece_end - piece);
+    const char *pos = nullptr;
+    if (with_case) {
+      pos = static_cast<const char *>(memchr(piece, c, piece_end - piece));
+    } else {
+      for (int i = 0; i < piece_end - piece; ++i) {
+        if (std::toupper(piece[i], std::locale()) == std::toupper(c, std::locale())) {
+          pos = piece + i;
+          break;
+        }
+      }
```

Comment:
Consider doing it via `strcasestr` like this:
```suggestion
      const char needle[] = {c, '\0'};
      pos = strcasestr(piece, needle);
```

Reply:
Done.